torch_xla/experimental/pjrt_backend.py

```diff
@@ -19,8 +20,17 @@ def _pjrt_rendezvous_handler(url: str,
   if not master_ip:
     master_ip = tpu.discover_master_worker_ip() if xr.device_type(
     ) == 'TPU' else 'localhost'
-
-  master_port = xu.getenv_as('MASTER_PORT', int, 12355)
+  # Workaround for Neuron to prevent socket in use error with torchrun
+  # MASTER_PORT is set by torchrun and will clash with the port here
+  if dist.is_torchelastic_launched() and xr.device_type() == 'NEURON':
+    master_port = xu.getenv_as('NEURON_PJRT_MASTER_PORT', int, 12355)
+    # In the case of torchrun pjrt.world_size() is not yet initialized
```

Comment:
Why doesn't `xr.world_size` work here? The runtime will always be initialized before this function returns, at least by `xr.global_ordinal`.

Reply:
I'm not fully sure, but any time I printed the world size (on PT 2.0) it would always be 1

Reply:
I believe it has to do with the fact that the replication is not set here before we call world_size() since we set the replication in xla_backend.py

Reply:
Actually, this should be `world_size`, because the `torch.distributed` world size should be 1 when doing single-process debugging (either not calling `xmp.spawn` or passing `nprocs=1`).

Getting a world size of 1 sounds suspicious to me. `xmp.spawn` will have already `set_replication` for you before you get to `init_process_group`, so [this case](https://github.com/pytorch/xla/blob/f06c31f5685fd537e9f220f3fdae8116d0bd0c32/torch_xla/runtime.py#L127-L128) will be false. Can you make sure your implementation of `PjRtClient::devices` is returning the correct set of devices, i.e. _all_ devices across all hosts/processes? `global_device_count` returns the [length](https://github.com/pytorch/xla/blob/f06c31f5685fd537e9f220f3fdae8116d0bd0c32/torch_xla/runtime.py#L121-L121) of [this list](https://github.com/pytorch/xla/blob/f06c31f5685fd537e9f220f3fdae8116d0bd0c32/torch_xla/csrc/runtime/pjrt_computation_client.cc#L708-L710)

Reply:
This is specifically for torchrun where it's needed